tests/run/reduce_pickle.pyx

```diff
@@ -300,5 +300,5 @@ if sys.version_info[:2] >= (3, 5):
         True
         >>> bound_method = pickle.loads(pickle.dumps(MyClass().my_method))
```

Comment:
And now that the method takes an argument, you could even call `bound_method(42)` here and see if it does the right thing.

Reply:
Yes. I encountered an obscure failure in the CI that I cannot reproduce locally after https://github.com/cython/cython/pull/2969/commits/1e2c7ee88fc6eb9b386d08642b1fdb6c81e997a5, so I am enriching the tests very incrementally to figure out what could have cause the failure. As soon as this turns green I'll add an assertion on the `method`'s result.

Reply:
It's looking good now, so I'll add the assertion.